### PR TITLE
Dont call toString on weld proxies for traces

### PR DESCRIPTION
**Pull request by benjamin-confino**

### CODE REVIEW

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#build

File: dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/inject/InjectInjectionBinding.java [hunk 1]
```diff
@@ -161,8 +170,8 @@ public Object getInjectionObject(Object targetObject,
         {
 
             if (CDIUtils.isInjectionFailureIgnored()){
-                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
-                    Tr.debug(tc, "com.ibm.ws.cdi.ignoreInjectionFailure = true, ignoring InjectionException", ex);
+                if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) {
+                    Tr.exit(tc, "com.ibm.ws.cdi.ignoreInjectionFailure = true, ignoring InjectionException", ex);
```
Comment (tevans78) [inline dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/inject/InjectInjectionBinding.java]: This exit should say it is returning null

File: dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/inject/InjectInjectionBinding.java [hunk 2]
```diff
@@ -69,12 +70,14 @@ public boolean isResolved() {
      */
     @SuppressWarnings("unchecked")
     @Override
+    @Trivial //This method is not actually trivial, but to avoid calling toString on a weld proxy as part of trace I will write the trace statements manually
```
[the remark this answers is not in the document]
Comment (benjamin-confino) [reply dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/inject/InjectInjectionBinding.java]: I asked on slack and was told it was not. 

We could take the statements out and test it.

File: dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/inject/InjectInjectionObjectFactory.java [hunk 3]
```diff
@@ -160,6 +161,7 @@ public static Object getObjectInstance(InjectInjectionBinding iBinding, Object t
 
             Object injectionObject = AccessController.doPrivileged(new PrivilegedAction<Object>() {
                 @Override
+                @Trivial //This method is not actually trivial, but to avoid calling toString on a weld proxy as part of trace I will write the trace statements manually
```
Comment (tevans78) [inline dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/inject/InjectInjectionObjectFactory.java]: You've added exit but no entry ... although not sure if it needs either in a doPriv

File: dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/inject/InjectInjectionObjectFactory.java [hunk 4]
```diff
@@ -192,13 +196,19 @@ public Object run() {
...
             Object[] referencesArray = references.toArray(new Object[references.size()]);
             if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) {
-                Tr.exit(tc, "getObjectInstance", referencesArray);
+                StringBuilder exitStringBuilder = new StringBuilder("[");
+                for (Object object : referencesArray){ 
+                    exitStringBuilder.append(Util.identity(object));
+                    exitStringBuilder.append(", ");
+                }
+                String exitString = exitStringBuilder.toString();
+                Tr.exit(tc, "getObjectInstance", exitString.substring(0,exitString.length()-2) + "]");
```
Comment (tevans78) [inline dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/inject/InjectInjectionObjectFactory.java]: better to move the string manipulation logic outside of the Tr.exit call, add an extra line above

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#build

**Comment by garypicher** [comment]
The latest personal build looks good - failures all seem to be slowness / timeout related, so I'll merge it.

**Pull request merged**